docs/en/integrations/data-ingestion/data-formats/csv-tsv.md

```diff
@@ -0,0 +1,377 @@
+
+
+### CSV files with headers
+
+Suppose our [CSV file has headers](assets/data_small_headers.csv) in it:
+
+```bash
+> head data-small-headers.csv
+"path","month","hits"
+"Akiba_Hebrew_Academy","2017-08-01",241
+"Aegithina_tiphia","2018-02-01",34
+```
+
+To import data from this file, we can use [CSVWithNames](/docs/en/interfaces/formats.md/#csvwithnames) format:
+
+```bash
+clickhouse-client -q "INSERT INTO sometable FORMAT CSVWithNames" < data_small_headers.csv
+```
+
+In this case, ClickHouse skips the first row while importing data from the file.
+
+
+### CSV files with custom delimiters
+
+In case the CSV file uses other than comma delimiter, we can use the [format_csv_delimiter](/docs/en/operations/settings/settings.md/#format_csv_delimiter) option to set the relevant symbol:
+
+
+```sql
+SET format_csv_delimiter = ';'
+```
+
+Now, when we import from a CSV file, `;` symbol is going to be used as a delimiter instead of a comma.
+
+
+### Skipping lines in a CSV file
+
+Sometimes, we might skip a certain number of lines while importing data from a CSV file. This can be done using [input_format_csv_skip_first_lines](/docs/en/operations/settings/settings.md/#input_format_csv_skip_first_lines) option:
+
+
+```sql
+SET input_format_csv_skip_first_lines = 10
+```
+
+In this case, we’re going to skip the first ten lines from the CSV file:
+
+```sql
+SELECT count(*)
+FROM file('data-small.csv', CSV)
+
+┌─count()─┐
+│     990 │
+└─────────┘
+```
+
+The [file](assets/data_small.csv) has 1k rows, but ClickHouse loaded only 990 since we’ve asked to skip the first 10.
+
+
+### Treating NULL values in CSV files
+
+Null values can be encoded differently depending on the application that generated the file. By default, ClickHouse uses `\N` as a Null value in CSV. But we can change that using the [format_csv_null_representation](/docs/en/operations/settings/settings.md/#format_tsv_null_representation) option.
+
+Suppose we have the following CSV file:
+
+```bash
+> cat nulls.csv
+Donald,90
+Joe,Nothing
+Nothing,70
+```
+
+If we load data from this file, ClickHouse will treat `Nothing` as a String (which is correct):
+
+```sql
+SELECT *
+FROM file('nulls.csv');
+
+┌─c1──────┬─c2──────┐
+│ Donald  │ 90      │
+│ Joe     │ Nothing │
+│ Nothing │ 70      │
+└─────────┴─────────┘
+```
+
+If we want ClickHouse to treat `Nothing` as `NULL`, we can define that using the following option:
+
+```sql
+SET format_csv_null_representation = 'Nothing'
+```
+
+Now we have `NULL` where we expect it to be:
+
+```sql
+SELECT *
+FROM file('nulls.csv');
+
+┌─c1─────┬─c2───┐
+│ Donald │ 90   │
+│ Joe    │ ᴺᵁᴸᴸ │
+│ ᴺᵁᴸᴸ   │ 70   │
+└────────┴──────┘
+```
+
+
+## TSV (Tab-separated) files
+
+Tab-separated data format is widely used, but not limited to, to migrate data between systems or nodes. To load data from a [TSV file](assets/data_small.tsv) to ClickHouse, [TabSeparated](/docs/en/interfaces/formats.md/#tabseparated) format is used:
```

Comment:
I drop `, but not limited to,` it doesn't read quite right grammatically, maybe

maybe 
" Tab-separated data format is widely used as a data interchange format."

Reply:
**the** format is used:

Reply:
fixed